### PR TITLE
E2E tests in Github

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

**Review by shnela — CHANGES_REQUESTED** [review]

File: e2e_tests/utils.py [hunk 1]
```diff
@@ -80,3 +83,17 @@ def image_to_png(*, image: Image) -> PngImageFile:
     image.save(png_buf, format="png")
     png_buf.seek(0)
     return PngImageFile(png_buf)
+
+
+def a_project_name(project_slug: str):
+    project_name = f"e2e-{datetime.now().strftime('%Y%m%d-%H%M')}-{project_slug}"
+    project_key = project_slug.replace("-", "")[:10].upper()
+    project_key = "".join(random.sample(project_key, len(project_key)))
```
[the remark this answers is not in the document]
Comment (Raalsky) [reply e2e_tests/utils.py:91]: > starting key with digit
Good point!
> Additoinally if we want to generate key by shuffling old one, we should do explicit shuffle ;)
I'm not getting it 😢

File: .github/workflows/full.yml [hunk 2]
```diff
@@ -16,9 +18,10 @@ jobs:
       - name: Lint
         uses: ./.github/actions/lint
 
-  integration:
+  test:
```
Comment (Raalsky) [inline .github/workflows/full.yml:21]: Add comment

File: .github/workflows/full.yml [hunk 3]
```diff
@@ -27,34 +30,66 @@ jobs:
...
+    env:
+      WORKSPACE_NAME: e2e-tests
+      BUCKET_NAME: ${{ secrets.E2E_BUCKET_NAME }}
+      AWS_ACCESS_KEY_ID: ${{ secrets.E2E_AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.E2E_AWS_SECRET_ACCESS_KEY }}
+      USER_USERNAME: ${{ secrets.E2E_USER_USERNAME }}
+      NEPTUNE_API_TOKEN: ${{ secrets.E2E_NEPTUNE_API_TOKEN }}
+      ADMIN_USERNAME: ${{ secrets.E2E_ADMIN_USERNAME }}
+      ADMIN_NEPTUNE_API_TOKEN: ${{ secrets.E2E_ADMIN_NEPTUNE_API_TOKEN }}
+      NEPTUNE_RETRIES_TIMEOUT: 600
```
Comment (Raalsky) [inline .github/workflows/full.yml:42]: Add comment

File: e2e_tests/conftest.py [hunk 4]
```diff
@@ -14,52 +14,75 @@
...
+        user=user,
+    )
+
+
+@pytest.fixture(scope="session")
+def container(request, environment):
     if request.param == "project":
-        project = neptune.init_project()
+        project = neptune.init_project(name=environment.project)
+        project.wait()
```
Comment (Raalsky) [inline e2e_tests/conftest.py:69]: To remove

**Review by shnela — APPROVED** [review]

**Pull request merged**